_posts/2018-02-16-Java-Time2.md

```diff
@@ -0,0 +1,1564 @@
+            enterprise.setName(enterpriseName); // 10
+            me.createChild(enterprise); // 11
+        } catch (RestException e) {
+            e.printStackTrace();
+        }
+        
+        return enterprise;
+    }
+    
+    public void deleteEnterprise(Enterprise enterprise) { // 12
+        if (enterprise == null) {
+            System.out.println("Enterprise does not exist");
+        } else {
+            System.out.println("Enterprise " + enterprise.getName() + " will be deleted");
+            try {
+                enterprise.delete(); // 13
+            } catch (RestException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+}
+{% endhighlight %}
+
+Note that in the constructor at Line 1, there is a `VSDSession` argument.  A `VSDSsssion` object establishes a secure HTTP connection to a VSD instance.  It is similar to how you would create a connection to a database before running SQL queries.  Before calling methods in this class, we will assume that a VSD session has already been established and started by `NetworkBuilder`.
+
+In Line 2 of the `verifyEnterprise` method, we perform a search for an existing VSD Enterprise object that matches the name for our enterprise that is passed in via `NetworkDescriptor`. If the Enterprise already exists, then we return that Enterprise object.  If the Enterprise does not exist, then in Line 3 we call the `createEnterprise` method to create and return a new Enterprise object.
+
+In the `fetchEnterprise` method at Line 4, we instantiate a `Me` object to serve as the root for all of our enterprise specific network objects.  Line 5 sets up a search filter based on the name of our enterprise.  In Line 6, we create an instance of `EnterprisesFetcher`.  The VSPK provides "Fetchers" for all entity types in the VSD to offer convenience methods for operating on collections of VSD objects.  Line 7 calls the `getFirst` method of the Fetcher to retrieve the Enterprise object matching the name in the search filter.
+
+Line 8 of the `createEnterprise` method is similar to Line 4 of the `fetchEnterprise` method.  It provides us with a 'Me' root object onto which we can attach a new Enterprise object.  In Line 9, we call the constructor of the VSPK Enterprise class to create a brand new Enterprise object and then in Line 10 we assign our enterprise name.  Finally, in Line 11, we create a new child Enterprise object.
+
+The method `deleteEnterprise` will be covered later when we discuss how to remove our network layout from the VSD.
+
+# Adding the Enterprise to our Network
+
+Here is the updated version of the `NetworkBuilder` class that integrates with `NetworkEnterpriseHandler`.
+
+{% highlight java %}
+import java.util.List;
+import net.nuagenetworks.bambou.RestException;
+import net.nuagenetworks.vspk.v5_0.Enterprise;
+import net.nuagenetworks.vspk.v5_0.VSDSession;
+
+public class NetworkBuilder {
+    private String url = "https://135.228.4.108:8443"; // 1

```

Comment:
Don't use a 135.... IP in documentation, use one from https://tools.ietf.org/html/rfc5737

Suggestion: `198.51.100.108`